src/ralph/backends/http/lrs.py

```diff
@@ -0,0 +1,277 @@
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        base_url: str = lrs_settings.BASE_URL,
+        username: str = lrs_settings.USERNAME,
+        password: str = lrs_settings.PASSWORD,
+        headers: LRSHeaders = lrs_settings.HEADERS,
+        status_endpoint: str = lrs_settings.STATUS_ENDPOINT,
+        statements_endpoint: str = lrs_settings.STATEMENTS_ENDPOINT,
+    ):
+        """Instantiates the LRS client.
+
+        Args:
+            base_url (AnyHttpUrl): LRS server URL.
+            username (str): Basic auth username for LRS authentication.
+            password (str): Basic auth password for LRS authentication.
+            headers (dict): Headers defined for the LRS server connection.
+            status_endpoint (str): Endpoint used to check server status.
+            statements_endpoint (str): Default endpoint for LRS statements resource.
+        """
+        self.base_url = parse_obj_as(AnyHttpUrl, base_url)
+        self.auth = (username, password)
+        self.headers = headers
+        self.status_endpoint = status_endpoint
+        self.statements_endpoint = statements_endpoint
+
+    def status(self):
+        """Implements HTTP backend check for server status."""
+        status_url = urljoin(self.base_url, self.status_endpoint)
+
+        try:
+            response = httpx.get(status_url)
+            response.raise_for_status()
+        except RequestError:
+            msg = "Unable to request the server"
+            logger.error(msg)
+            return HTTPBackendStatus.AWAY
+        except HTTPStatusError:
+            msg = "Response raised an HTTP status of 4xx or 5xx"
+            logger.error(msg)
+            return HTTPBackendStatus.ERROR
+
+        return HTTPBackendStatus.OK
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        # noqa: D205, D415
+        """LRS HTTP backend does not support `list` method,
+        calling this method will raise an error.
+        """
+        msg = "LRS HTTP backend does not support `list` method, cannot list from %s"
+        logger.error(msg, target)
+        raise NotImplementedError(msg % target)
+
+    @enforce_query_checks
+    def read(  # pylint: disable=too-many-arguments
+        self,
+        query: Union[str, LRSQuery] = None,
+        target: str = None,
+        chunk_size: Union[None, int] = 500,
+        raw_output: bool = False,
+        ignore_errors: bool = False,
+    ) -> Iterator[Union[bytes, dict]]:
+        """Get statements from LRS `target` endpoint.
+
+        The `read` method defined in the LRS specification returns `statements` array
+        and `more` IRL. The latter is required when the returned `statement` list has
+        been limited.
+
+        Args:
+            query (str, LRSQuery):  The query to select records to read.
+            target (str): Endpoint from which to read data (e.g. `/statements`).
```

Comment:
perhaps include a helper for users that forget the slash ? (seems like it could be an easy mistake to make)

Reply:
The `target` parameter is used in the `ParseResult` method. Whether the slash is missing, the target is correctly parsed for the URL